dingo/backup/redundant_facet_removal_backup.py

```diff
@@ -0,0 +1,642 @@
+
+    max_biomass_flux_vector, max_biomass_objective = fast_fba(lb, ub, S, c)
+    val = -np.floor(max_biomass_objective / tol) * tol * opt_percentage / 100
+
+    b_res = []
+    A_res = np.empty((0, n), float)
+    beq_res = np.array(beq)
+
+    try:
+        with gp.Env(empty=True) as env:
+            env.setParam("OutputFlag", 0)
+            env.start()
+
+            with gp.Model(env=env) as model:
+                x = model.addMVar(
+                    shape=n,
+                    vtype=GRB.CONTINUOUS,
+                    name="x",
+                    lb=lb,
+                    ub=ub,
+                )
+
+                Aeq_sparse = sp.csr_matrix(S)
+                A_sparse = sp.csr_matrix(np.array(-c))
+                b_sparse = np.array(val)
+
+                b = np.array(b)
+                beq = np.array(beq)
+
+                model.addMConstr(Aeq_sparse, x, "=", beq, name="c")
+                model.update()
+                model.addMConstr(A_sparse, x, "<", [val], name="d")
+                model.update()
+
+                model_iter = model.copy()
+
+                indices_iter = range(n)
+                removed = 1
+                offset = 1
+                facet_left_removed = np.zeros((1, n), dtype=bool)
+                facet_right_removed = np.zeros((1, n), dtype=bool)
+
+                while removed > 0 or offset > 0:
+                    removed = 0
+                    offset = 0
+                    indices = indices_iter
+                    indices_iter = []
+
+                    Aeq_sparse = sp.csr_matrix(Aeq_res)
+                    beq = np.array(beq_res)
+
+                    b_res = []
+                    A_res = np.empty((0, n), float)
+                    for i in indices:
+                        objective_function = A[i]
+
+                        redundant_facet_right = True
+                        redundant_facet_left = True
+
+                        objective_function_max = np.asarray(
+                            [-x for x in objective_function]
+                        )
+
+                        # Call solve_lp_with_different_objectives to solve LP
+                        model_iter = solve_lp_with_different_objectives(
+                            model_iter, objective_function_max
+                        )
+
+                        status = model_iter.status
+                        if status == GRB.OPTIMAL:
+                            max_objective = -model_iter.objVal
+                        else:
+                            max_objective = ub[i]
+
+                        if not facet_right_removed[0, i]:
+                            ub_iter = ub.copy()
+                            ub_iter[i] = ub_iter[i] + 1
+
+                            # Call solve_lp_with_different_objectives to solve LP
+                            model_iter = solve_lp_with_different_objectives(
+                                model_iter, objective_function
+                            )
```

Comment:
plz change this to:
model_iter = solve_lp_with_different_objectives(
                                model_iter.copy(), objective_function
                            )

Reply:
ok